Import PropTypes from 'prop-types'

kylecesmat opened the pull request:
Fixes #92 

This PR adds support for React 15.5+ by including the `prop-types` package. This also upgrades react-codemirror & offers a few housekeeping items.

- Upgrade `react-codemirror@1.0.0`
- Include `prop-types` package as a dependency
- Remove support for React < 15.4
- Use yarn for dependency management
- Upgrade travis configuration, remove old node versions

ryan-roemer commented:
@kylecesmat -- If you're switching to yarn, you should update travis config too. Here's a (untested) suggestion:

```yaml
language: node_js

node_js:
  - "4"
  - "6"

sudo: false

branches:
  only:
    - master

cache: yarn

before_install:
  # GUI for real browsers.
  - export DISPLAY=:99.0
  - sh -e /etc/init.d/xvfb start

script:
  - yarn run check-ci

  # Prune deps to just production and ensure we can still build
  - yarn install --production --ignore-scripts --prefer-offline
  - yarn run build
```

kylecesmat commented:
@ryan-roemer thanks for the suggestion! I modified the travis configuration, and removed the commands we should no longer need, now that we're using yarn.

Would you be able to provide a review?

ryan-roemer commented:
@kylecesmat -- Restarting failed travis job on node6: https://travis-ci.org/FormidableLabs/component-playground/jobs/233708066

kylecesmat commented:
@ryan-roemer yeah I'm seeing this error locally now on `node6`, but runs fine with `node4`. I'm assuming one of our phantomjs modules needs to be updated - I'm timeboxed on this right now but will investigate later.

ryan-roemer commented:
@kylecesmat -- Feel free to comment out `- "6"` and place a ticket link there and ticket out "figure out phantom / update"

chrisbolin commented on .travis.yml:
```diff
@@ -11,16 +12,14 @@ branches:
+
 before_install:
   # GUI for real browsers.
-  - npm install -g npm@3
```
How is yarn installed on Travis with this setup?
